src/TorchVision/dsets/CelebA.cs

```diff
@@ -0,0 +1,276 @@
+            {
+                foreach (var (_, md5, filename) in file_list) {
+                    var fpath = Path.Combine(this.root, base_folder, filename);
+                    var ext = Path.GetExtension(filename);
+                    // Allow original archive to be deleted (zip and 7z)
+                    // Only need the extracted images
+                    if (ext != ".zip" && ext != ".7z" && !check_integrity(fpath, md5)) {
+                        return false;
+                    }
+                }
+
+                // Should check a hash of the images
+                return Directory.Exists(Path.Combine(this.root, base_folder, "img_align_celeba"));
+            }
+
+            public void download()
+            {
+                if (this._check_integrity()) {
+                    Console.WriteLine("Files already downloaded and verified");
+                    return;
+                }
+
+                foreach (var (file_id, md5, filename) in file_list) {
+                    download_file_from_google_drive(file_id, Path.Combine(this.root, base_folder), filename, md5);
+                }
+
+                extract_archive(Path.Combine(this.root, base_folder, "img_align_celeba.zip"));
+            }
+
+            private struct CSV
+            {
+                public string[]? header;
+                public string[] index;
+                public torch.Tensor data;
+                public CSV(string[]? header, string[] index, torch.Tensor data)
+                {
+                    this.header = header;
+                    this.index = index;
+                    this.data = data;
+                }
+            }
+        }
+    }
+
+    public static partial class torchvision
+    {
+        public static partial class datasets
+        {
+            /// <summary>
+            /// `Large-scale CelebFaces Attributes (CelebA) Dataset http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html Dataset.
+            /// </summary>
+            /// <param name="root">Root directory where images are downloaded to.</param>
+            /// <param name="split">One of {'train', 'valid', 'test', 'all'}.
+            /// Accordingly dataset is selected.</param>
+            /// <param name="target_type">Type of target to use, ``attr``, ``identity``, ``bbox``,
+            /// or ``landmarks``.</param>
+            /// <param name="transform">A function/transform that  takes in an PIL image
+            /// and returns a transformed version.</param>
+            /// <param name="target_transform">A function/transform that takes in the
+            /// target and transforms it.</param>
+            /// <param name="download">If true, downloads the dataset from the internet and
+            /// puts it in root directory. If dataset is already downloaded, it is not
+            /// downloaded again.</param>
+            public static Dataset CelebA(
+                string root,
+                string split = "train",
```

Comment:
Could this be an enum instead of a string? That's one area where we usually diverge from PyTorch.

Reply:
Fixed by 0b49104c07c7b9e3e37b13e267a3c9e41bee9ca1